src/io/filewatcher.rs

```diff
@@ -17,11 +17,17 @@ pub struct FileWatcher {
     lines_buf: VecDeque<String>,
     /// Size of the file (in bytes) when it was last read
     last_size: usize,
+    /// The file being watched
+    file: File,
 }
 
 impl FileWatcher {
     pub fn new(path: PathBuf) -> Result<FileWatcher> {
-        let file = FileWatcher::open_file(&path).context("Failed to open file to watch.")?;
+        let file = OpenOptions::new()
+            .read(true)
+            .write(false)
+            .open(&path)
+            .context("Failed to open file to watch.")?;
```

Comment:
You will have to seek to `last` here when initially opening the file otherwise it will read in all the old data, which may have been from previous games for example.